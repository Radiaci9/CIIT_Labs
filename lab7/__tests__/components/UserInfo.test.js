import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import UserInfo from '../../src/components/UserInfo'

describe('<UserInfo />', () => {
  describe('render()', () => {
    test('renders the component', () => {
        const user = {
            avatar: null,
            login: 'login',
            name: 'name',
            bio: 'bio',
            company: 'company',
            location: 'location',
            email: 'email',
            blog: 'blog'
        }
        const wrapper = shallow(<UserInfo user={user}/>)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('set login', () => {
    test('successfully set login', () => {
        const user = {
            avatar: null,
            login: 'login',
            name: 'name',
            bio: 'bio',
            company: 'company',
            location: 'location',
            email: 'email',
            blog: 'blog'
        }
        const wrapper = shallow(<UserInfo user={user}/>)
        expect(wrapper.find('[name="login"]').text()).toEqual('login') 
    })
  })

  describe('loading user', () => {
    test('user not found', () => {
        const user = {
            avatar: null,
            login: null,
            name: null,
            bio: null,
            company: null,
            location: null,
            email: null,
            blog: null
        }
        const wrapper = shallow(<UserInfo user={user}/>)
        expect(wrapper.find('img')).toHaveLength(1) 
    })
  })
})
