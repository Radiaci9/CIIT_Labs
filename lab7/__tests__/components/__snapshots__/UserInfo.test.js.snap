// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> render() renders the component 1`] = `
<div>
  <img
    alt="avatar"
    src={null}
    style={
      Object {
        "borderRadius": "15px",
        "height": "250px",
        "width": "250px",
      }
    }
  />
  <h2
    className="nullMargin"
    name="name"
  >
    name
  </h2>
  <p
    className="nullMargin"
    name="login"
  >
    login
  </p>
  <p
    name="bio"
    style={
      Object {
        "width": "250px",
      }
    }
  >
    bio
  </p>
  <hr
    style={
      Object {
        "marginLeft": 0,
        "width": "250px",
      }
    }
  />
  <LabelWithIcon
    iconSrc="../img/organization.svg"
    label="company"
    link={false}
  />
  <LabelWithIcon
    iconSrc="../img/location.svg"
    label="location"
    link={false}
  />
  <LabelWithIcon
    iconSrc="../img/send.svg"
    label="email"
    link={true}
  />
  <LabelWithIcon
    iconSrc="../img/link.svg"
    label="blog"
    link={true}
  />
</div>
`;
